libs/src/sanityChecks/rolloverNodes.js

```diff
@@ -1,69 +1,88 @@
+const _ = require('lodash')
+
 const Utils = require('../utils')
 const CreatorNode = require('../services/creatorNode')
 
 const THREE_SECONDS = 3000
 const MAX_TRIES = 3
 
-/** Check if the user's primary creator node is healthy */
-const checkPrimaryHealthy = async (libs, primary, tries) => {
-  const healthy = await Utils.isHealthy(primary)
+const nodeIsHealthy = async (endpoint, tries) => {
+  const healthy = await Utils.isHealthy(endpoint)
   if (healthy) return healthy
   else {
-    if (tries === 0) {
-      return false
-    }
+    if (tries === 0) return false
     await Utils.wait(THREE_SECONDS)
-    return checkPrimaryHealthy(libs, primary, tries - 1)
+    return nodeIsHealthy(endpoint, tries - 1)
   }
 }
 
-/** Gets new endpoints from a user's secondaries */
-const getNewPrimary = async (libs, secondaries) => {
+const getUpToDateSecondary = async (libs, secondaries) => {
```

Comment:
Changed name to reflect usage. We use this function not only to select new Primary but to find any up to date secondary.
Also replaced throw Error with return null to reflect usage.